### ISSUE TITLE
OSX installation issue

### ISSUE THREAD

**Issue by ashermancinelli** [dzaima/CBQN#7]
Based on docs from the [main BQN page](https://github.com/mlochbaum/BQN/blob/master/running.md) and this repo, I can't seem to run what's built:

```console
$ make # builds with a couple warnings
$ rlwrap ./BQN
Error: JIT: Refusing to run with CBQN code outside of the 32-bit address range
2: source unknown
1: source unknown
0: source unknown
```

**Comment by dzaima** [comment]
oh right, still have to fix this. The temporary solution is either changing [this](https://github.com/dzaima/CBQN/blob/084eaaecbd26133891c6f1f78491efdc093aff68/src/vm.h#L6) definition to `-1`, or building with `make t=nojit f='-O3 -DJIT_START=-1' c`.

The JIT wants to be able to allocate near the source code to be able to use relative addresses, but for whatever reason OSX doesn't allocate the program at the

**Comment by dzaima** [comment]
Pushed a potential solution, would like to hear if it works for you!

**Comment by ashermancinelli** [comment]
Thanks for being so quick!! I'll try to test it sometime today.

**Comment by ashermancinelli** [comment]
☹️ 
```console
jit/nvm.c
In file included from src/jit/nvm.c:5:
src/jit/nvm_x86_64.c:49:103: error: use of undeclared identifier 'MAP_FIXED_NOREPLACE'
    void* c = mmap((void*)loc, sz, PROT_READ|PROT_WRITE|PROT_EXEC, MAP_NORESERVE|MAP_PRIVATE|MAP_ANON|MAP_FIXED_NOREPLACE, -1, 0);
                                                                                                      ^
1 error generated.
make[1]: *** [obj/o3/nvm.o] Error 1
make[1]: *** Waiting for unfinished jobs....
1 warning generated.
make: *** [o3] Error 2
```

**Comment by ashermancinelli** [comment]
Closed with commit eca2a7e
